src/main/java/seedu/address/logic/commands/FindCommand.java

```diff
@@ -18,25 +23,68 @@
             + "the specified keywords (case-insensitive) and displays them as a list with index numbers.\n"
             + "Parameters: KEYWORD [MORE_KEYWORDS]...\n"
             + "Example: " + COMMAND_WORD + " alice bob charlie";
+    public static final String WARNING_INVALID_CONTEXT = "This command can only be used"
+            + "while viewing the list of activities/contacts.";
+    private final String searchTerm;
+    private final String[] keywords;
 
-    private final NameContainsKeywordsPredicate predicate;
-
-    public FindCommand(NameContainsKeywordsPredicate predicate) {
-        this.predicate = predicate;
+    public FindCommand(String[] keywords, String searchTerm) {
+        this.keywords = keywords;
+        this.searchTerm = searchTerm;
+        System.out.print("Keywords: ");
+        for (String a : keywords) {
+            System.out.print(a + ",");
+        }
+        System.out.println("");
+        System.out.println("searchTerm: " + searchTerm);
     }
 
     @Override
-    public CommandResult execute(Model model) {
+    public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
-        model.updateFilteredPersonList(predicate);
-        return new CommandResult(
-                String.format(Messages.MESSAGE_PERSONS_LISTED_OVERVIEW, model.getFilteredPersonList().size()));
+        ContextType type = model.getContext().getType();
+        switch (type) {
+        case LIST_CONTACT:
+            model.updateFilteredPersonList(new NameContainsKeywordsPredicate(Arrays.asList(keywords)));
+            return new CommandResult(String.format(Messages.MESSAGE_FOUND_BY_KEYWORD,
+                    model.getFilteredPersonList().size(),
+                    pluraize("contact", model.getFilteredPersonList().size()),
+                    searchTerm));
+
+        case LIST_ACTIVITY:
+            model.updateFilteredActivityList(new TitleContainsKeywordsPredicate(Arrays.asList(keywords)));
+
+            return new CommandResult(String.format(Messages.MESSAGE_FOUND_BY_KEYWORD,
+                    model.getFilteredActivityList().size(),
+                    pluraize("activity", model.getFilteredActivityList().size()),
+                    searchTerm));
+        default:
+            throw new CommandException(WARNING_INVALID_CONTEXT);
+        }
     }
 
     @Override
     public boolean equals(Object other) {
         return other == this // short circuit if same object
                 || (other instanceof FindCommand // instanceof handles nulls
-                && predicate.equals(((FindCommand) other).predicate)); // state check
+                && Arrays.equals(this.keywords, ((FindCommand) other).keywords)
+                && searchTerm.equals(((FindCommand) other).searchTerm));
+    }
+
+    /**
+     * Helper method that helps to pluraize given string, depending on the number given.
+     */
+    private String pluraize(String input, int count) {
```

Comment:
I think it's spelled `pluralize` instead of `pluraize`.

Reply:
Fixed spelling error.